### PR TITLE
add hbase backend to Travis CI tests

### PR DESCRIPTION
**Pull request by zhoney**
Change-Id: I01cd1b8d764249f2f6570129409f889fd8e06ceb

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,8 +17,18 @@ addons:
       - mysql-server
       - mysql-client
 
+cache:
+  directories:
+  - $HOME/downloads
```
Comment (javeme) [inline .travis.yml]: align

File: .travis/install-hbase.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# download hbase
+if [ ! -f $HOME/downloads/hbase-2.0.1-bin.tar.gz ]; then
+  sudo wget -O $HOME/downloads/hbase-2.0.1-bin.tar.gz http://apache.cs.utah.edu/hbase/2.0.1/hbase-2.0.1-bin.tar.gz;
+fi
+
+# decompress hbase
+sudo mv $HOME/downloads/hbase-2.0.1-bin.tar.gz hbase-2.0.1-bin.tar.gz && tar xzf hbase-2.0.1-bin.tar.gz
+
+# config hbase
+sudo rm -f hbase-2.0.1/conf/hbase-site.xml && sudo mv .travis/hbase-site.xml hbase-2.0.1/conf
```
Comment (javeme) [inline .travis/install-hbase.sh]: use cp instead of mv

File: .travis/install-hbase.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+# download hbase
+if [ ! -f $HOME/downloads/hbase-2.0.1-bin.tar.gz ]; then
+  sudo wget -O $HOME/downloads/hbase-2.0.1-bin.tar.gz http://apache.cs.utah.edu/hbase/2.0.1/hbase-2.0.1-bin.tar.gz;
+fi
+
+# decompress hbase
+sudo mv $HOME/downloads/hbase-2.0.1-bin.tar.gz hbase-2.0.1-bin.tar.gz && tar xzf hbase-2.0.1-bin.tar.gz
```
Comment (javeme) [inline .travis/install-hbase.sh]: use cp instead of mv

File: .travis.yml [hunk 4]
```diff
@@ -17,8 +17,18 @@ addons:
...
+cache:
+  directories:
+    - $HOME/downloads
+
 install: mvn compile
 
+before_script:
+  - |
+    if [ "$BACKEND" == "hbase" ]; then
+      .travis/install-hbase.sh
```
Comment (javeme) [inline .travis.yml]: prefer to move .travis to hugegraph-dist/src/assembly/travis

File: .travis/install-hbase.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+# download hbase
+if [ ! -f $HOME/downloads/hbase-2.0.1-bin.tar.gz ]; then
+  sudo wget -q -O $HOME/downloads/hbase-2.0.1-bin.tar.gz http://apache.cs.utah.edu/hbase/2.0.1/hbase-2.0.1-bin.tar.gz
+fi
+
+# decompress hbase
+sudo cp $HOME/downloads/hbase-2.0.1-bin.tar.gz hbase-2.0.1-bin.tar.gz && tar xzf hbase-2.0.1-bin.tar.gz
+
+# config hbase
+sudo rm -f hbase-2.0.1/conf/hbase-site.xml && sudo cp .travis/hbase-site.xml hbase-2.0.1/conf
```
Comment (javeme) [inline .travis/install-hbase.sh]: add a var for ".travis"

File: hugegraph-dist/src/assembly/travis/install-hbase.sh [hunk 6]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+set -ev
+
```
Comment (javeme) [inline hugegraph-dist/src/assembly/travis/install-hbase.sh]: define a var like BASE_DIR for `dirname $0`

File: hugegraph-dist/src/assembly/travis/hbase-site.xml [hunk 7]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0"?>
+<?xml-stylesheet type="text/xsl" href="configuration.xsl"?>
+<!--
```
Comment (javeme) [inline hugegraph-dist/src/assembly/travis/hbase-site.xml]: remove License

File: .travis/hbase-site.xml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    </property>
+    <property>
+        <name>zookeeper.session.timeout</name>
+        <value>1200000</value>
+    </property>
+    <property>
+        <name>hbase.zookeeper.property.tickTime</name>
+        <value>6000</value>
+    </property>
+</configuration>
```
[the remark this answers is not in the document]
Comment (zhoney) [reply .travis/hbase-site.xml]: done

File: .travis.yml (hunk 1 above)
Comment (zhoney) [reply .travis.yml]: done

File: .travis/install-hbase.sh (hunk 2 above)
Comment (zhoney) [reply .travis/install-hbase.sh]: done

File: .travis/install-hbase.sh (hunk 3 above)
Comment (zhoney) [reply .travis/install-hbase.sh]: done

File: .travis.yml (hunk 4 above)
Comment (zhoney) [reply .travis.yml]: done

File: hugegraph-dist/src/assembly/travis/install-hbase.sh (hunk 6 above)
Comment (zhoney) [reply hugegraph-dist/src/assembly/travis/install-hbase.sh]: done

File: hugegraph-dist/src/assembly/travis/hbase-site.xml (hunk 7 above)
Comment (zhoney) [reply hugegraph-dist/src/assembly/travis/hbase-site.xml]: done

**Pull request merged**